advanced/Scripts/piholeDebug.sh

```diff
@@ -1012,53 +1041,66 @@ analyze_gravity_list() {
   IFS="$OLD_IFS"
 }
 
+gravity_log_checks_and_obfuscation() {
+  local error_to_check_for
+  local line_to_obfuscate="${1}"
+  local obfuscated_line
+  # A common error in the pihole.log is when there is a non-hosts formatted file
+  # that the DNS server is attempting to read.  Since it's not formatted
+  # correctly, there will be an entry for "bad address at line n"
+  # So we can check for that here and highlight it in red so the user can see it easily
+  error_to_check_for=$(echo ${line_to_obfuscate}} | grep 'bad address at')
+  # Some users may not want to have the domains they visit sent to us
+  # To that end, we check for lines in the log that would contain a domain name
+  line_to_obfuscate=$(echo ${line_to_obfuscate} | grep ': query\|: forwarded\|: reply\|: cached \|: /etc/pihole/gravity\.list')
+  # If the variable contains a value, it found an error in the log
+  if [[ -n ${error_to_check_for} ]]; then
+    # So we can print it in red to make it visible to the user
+    log_write "   ${CROSS} ${COL_RED}${line_to_obfuscate}${COL_NC} (${FAQ_BAD_ADDRESS})"
+  else
+    # If the variable is set to true,
+    if [[ ${OBFUSCATE} == "true" ]]; then
+      # check if there are domains in the log that should be obfuscated
+      if [[ -n ${line_to_obfuscate} ]]; then
+        # If there are, we need to use awk to replace only the domain name (the 6th field in the log)
+        # so we substitue the domain for the placeholder value
+        obfuscated_line=$(echo ${line_to_obfuscate} | awk -v placeholder="${OBFUSCATED_PLACEHOLDER}" '{sub($6,placeholder); print $0}')
```

Comment:
Will this replace *all* domains by the *same* dummy domain?

Reply:
Yeah.  And on second thought.  This PR will be better for a subsequent release.  I should add some options to turn it off and on